frontend/src/scenes/dashboard/dashboardLogic.js

```diff
@@ -319,5 +318,30 @@ export const dashboardLogic = kea({
                 setTimeout(() => actions.refreshDashboardItem(id), 1000)
             }
         },
+        reportUsage: async (payload, breakpoint) => {
+            await breakpoint(500) // Debounce to avoid noisy events from continuous navigation
+            const { created_at, name, is_shared, pinned } = payload
+            const properties = {
+                created_at,
+                name,
+                is_shared,
+                pinned,
+                sample_items_count: 0,
+                item_count: payload.items.length,
+                created_by_system: !payload.created_by,
+                watching_shared: !!props.shareToken,
+            }
+
+            for (const item of payload.items) {
```

Comment:
Not-actionable: We're adding a ton of cruft like this for analytics everywhere - the extra lines of code do make things less maintainable as it's not always clear what's real code and what's analytics.

If we were using lodash, we could get rid of most lines of code using it here using sums/groupBys. The arrays here are small enough that there isn't really a performance win writing long loops.